include/ocpp/common/message_queue.hpp

```diff
@@ -71,26 +64,78 @@ template <typename M> struct ControlMessage {
     std::promise<EnhancedMessage<M>> promise; ///< A promise used by the async send interface
     DateTime timestamp;                       ///< A timestamp that shows when this message can be sent
     MessageId initial_unique_id;
+    bool stall_until_accepted; // if true, message shall be sent only if registration status is accepted
 
     /// \brief Creates a new ControlMessage object from the provided \p message
-    explicit ControlMessage(const json& message);
+    explicit ControlMessage(const json& message, const bool stall_until_accepted = false);
 
     /// \brief Provides the unique message ID stored in the message
     /// \returns the unique ID of the contained message
     [[nodiscard]] MessageId uniqueId() const {
         return this->message[MESSAGE_ID];
     }
 
-    /// \brief Determine whether message is considered as transaction-related.
-    bool isTransactionMessage() const;
-
     /// \brief True for transactional messages containing updates (measurements) for a transaction
     bool isTransactionUpdateMessage() const;
+};
+
+/// \brief This can be used to distinguish the different queue types
+enum class QueueType {
+    Normal,
+    Transaction,
+    None,
+};
 
-    /// \brief Determine whether message is a BootNotification.
-    bool isBootNotificationMessage() const;
+/// \brief Indicates the transmission priority of a message that is being pushed to the message queue
+enum class MessageTransmissionPriority {
+    SEND_IMMEDIATELY,                        // message can be queued and can be send immediately
```

Comment:
Why the all caps? Would be better to keep it consistent I would say and leave them CamelCase like we do with all enum classes.